addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,1024 @@
+                    // it captures an infrared sequence from a remote control
+                    if (strippedMessage.contains("RCV_HEX")) {
+                        parseHexMessage(strippedMessage);
+                    }
+
+                    // IRTrans devices return a string starting with RCV_COM each time
+                    // it captures an infrared sequence from a remote control that is stored in the device's internal dB
+                    if (strippedMessage.contains("RCV_COM")) {
+                        parseIRDBMessage(strippedMessage);
+                    }
+
+                    byteCount = getByteCount(byteBuffer);
+
+                } else {
+                    logger.warn("Received some non-compliant garbage '{}' - Parsing is skipped",
+                            new String(byteBuffer.array()));
+                }
+            }
+        } catch (Exception e) {
+            logger.error("An exception occurred while reading bytebuffer '{}' : {}",
+                    DatatypeConverter.printHexBinary(byteBuffer.array()), e.getMessage(), e);
+        }
+    }
+
+    private Runnable pollingRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                if (socketChannel == null) {
+                    previousConnectionState = false;
+                    onConnectionLost();
+                } else {
+                    if (previousConnectionState == false && socketChannel.isConnected()) {
+                        previousConnectionState = true;
+                        onConnectionResumed();
+                    }
+
+                    if (previousConnectionState == true && !socketChannel.isConnected()
+                            && !socketChannel.isConnectionPending()) {
+                        previousConnectionState = false;
+                        onConnectionLost();
+                    }
+
+                    if (!socketChannel.isConnectionPending() && !socketChannel.isConnected()) {
+                        previousConnectionState = false;
+                        logger.debug("Disconnecting '{}' because of a network error", getThing().getUID().toString());
+                        disconnect(socketChannel);
+                        Thread.sleep(1000 * ((BigDecimal) getConfig().get(RECONNECT_INTERVAL)).intValue());
+                        establishConnection();
+                    }
+
+                    long stamp = System.currentTimeMillis();
+                    if (!InetAddress.getByName(((String) getConfig().get(IP_ADDRESS)))
+                            .isReachable(((BigDecimal) getConfig().get(PING_TIME_OUT)).intValue())) {
+                        logger.debug(
+                                "Ping timed out after '{}' milliseconds. Disconnecting '{}' because of a ping timeout",
+                                System.currentTimeMillis() - stamp, getThing().getUID().toString());
```

Comment:
You don't have to call toString it will be called automatically, but only when the logging-level actually applies. Can you check all occurrences